WordPress/gutenberg
# Block Library: Add new Comment Reply Link block

## Description


This PR implements a block which adds a link to reply to a specific comment. The implementation will follow the design explained in the linked issue.

Will close #30576.

## How has this been tested?




<details>
<summary><b>Typography, align and Color settings</b></summary>
</br>

1. Add a post comment block (using an existing comment ID).
2. Add an inner post comment reply link block.
3. Confirm that the color and typography options work.
4. Confirm that the text align option works.

**NOTE:** there is an issue related to global styles, check https://github.com/WordPress/gutenberg/pull/35774#issuecomment-952046182

</details>

<details>
<summary><b>Threads depth limit</b></summary>
</br>

1. Create a comment (comment 1).
2. Create another comment, replying the previous one (comment 2).
3. Go to site editor and, in the Single Post template, add a post comment block with the reply link for each comment.
4. In the frontend, go to a post and confirm that both render the correct reply link.
5. Go to Discussions Settings and change the "levels deep" setting to `2`.
6. Confirm that only comment 1 renders a reply link.
7. Go back to Discussions Settings and disable threaded comments entirely.
8. Confirm that none of the comments render a reply link.

</details>

<details>
<summary><b>Invalid or missing comment ID</b></summary>
</br>

1. Add a comment block and set an invalid comment ID.
2. In the frontend, confirm that the block is not rendered.

</details>

<details>
<summary><b>Only registered and logged in users can comment</b></summary>
</br>

1. In Discussions Settings, enable "Users must be registered and logged in to comment"
2. Add a comment block in the Single Post template, and set a valid comment ID.
3. Visit a post in incognito.
4. confirm that the rendered reply link shows a `Log in to reply` message or similar.

</details>

## Types of changes
New feature

## Checklist:
